iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using iox::roudi::RouDiEnvironment;
+class PoshRuntime_test : public Test
+{
+  public:
+    PoshRuntime_test()
+    {
+    }
+
+    virtual ~PoshRuntime_test()
+    {
+    }
+
+    virtual void SetUp()
+    {
+        internal::CaptureStdout();
+    };
+
+    virtual void TearDown()
+    {
+        std::string output = internal::GetCapturedStdout();
+        if (Test::HasFailure())
+        {
+            std::cout << output << std::endl;
+        }
+    };
+
+    void InterOpWait()
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+    }
+
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_runtime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::string<100> m_runnableName{"testRunnable"};
+    const iox::cxx::string<100> m_invalidRunnableName{"invalidRunnable,"};
+    static bool m_errorHandlerCalled;
+    const iox::cxx::string<100> m_runtimeName{"/sender"};
+};
+
+bool PoshRuntime_test::m_errorHandlerCalled{false};
+
+
+TEST_F(PoshRuntime_test, ValidAppName)
+{
+    std::string appName("/valid_name");
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(appName); });
+}
+
+
+TEST_F(PoshRuntime_test, AppNameLength_OutOfLimit)
+{
+    std::string tooLongName(100, 's');
```

Comment:
Please get rid of the magic number `100` here. The Maximum app name length is defined in `iceoryx_posh_types.hpp` in the variable `MAX_PROCESS_NAME_LENGTH`